projects/igniteui-angular/src/lib/grids/row-drag.directive.ts

```diff
@@ -0,0 +1,137 @@
+import { Directive, Input, OnDestroy, NgModule } from '@angular/core';
+import { IgxDragDirective } from '../directives/dragdrop/dragdrop.directive';
+import { IRowDragEndEventArgs, IRowDragStartEventArgs } from './grid-base.component';
+import { KEYS } from '../core/utils';
+import { fromEvent, Subscription } from 'rxjs';
+import { IgxRowComponent, IgxGridBaseComponent, IGridDataBindable } from './grid';
+
+
+const ghostBackgroundClass = 'igx-grid__tr--ghost';
+const gridCellClass = 'igx-grid__td';
+
+/**
+ * @hidden
+ */
+@Directive({
+    selector: '[igxRowDrag]'
+})
+export class IgxRowDragDirective extends IgxDragDirective implements OnDestroy {
+    private row: IgxRowComponent<IgxGridBaseComponent & IGridDataBindable>;
+    private subscription$: Subscription;
+    private _rowDragStarted = false;
+    public startDrag(event) {
+        this.onPointerDown(event);
+    }
+
+    @Input('igxRowDrag')
+    set data(val) {
+        this.row = val;
+    }
+
+    get data() {
+        return this.row;
+    }
+
+    public onPointerDown(event) {
+        event.preventDefault();
+        this._rowDragStarted = false;
+        super.onPointerDown(event);
+    }
+
+    public onPointerMove(event) {
+        super.onPointerMove(event);
+        if (this._dragStarted && !this._rowDragStarted) {
+            this._rowDragStarted = true;
```

Comment:
Confused, what's the point in `_rowDragStarted`?
Seems like you wanted to override the start emitter. You could've moved it to a base onStart method that you can override, no?